src/libs/datatableproviders/DataProvider.ts

```diff
@@ -0,0 +1,88 @@
+// define metadata structures
```

Comment:
we are starting to organize the ajax related code in libs to be under src/libs/ajax... So how about this folder gets moved to src/libs/ajax/workspaces.

Reply:
hm - I disagree with putting these under a "workspaces" directory, as that's pretty generic. I've moved to `src/libs/ajax/datatableproviders`. This is not a hill I will die on - if you feel strongly about using `src/libs/ajax/workspaces` instead of `src/libs/ajax/datatableproviders`, speak up and I will change again.

Reply:
Can it be `src/libs/ajax/data-table-providers` (with dashes) for consistency and easier reading?

Reply:
renamed `src/libs/ajax/datatableproviders` to `src/libs/ajax/data-table-providers`.

Reply:
this is quite workable.  thx!